handlers/handlers.go

```diff
@@ -0,0 +1,244 @@
+package handlers
+
+import (
+	"encoding/json"
+	"gin/models"
+	"github.com/gin-gonic/gin"
+	"github.com/olivere/elastic/v7"
+	"html"
+	"net/http"
+	"strconv"
+)
+
+const INDEX_NAME = "books"
+
+func CreateBook(c *gin.Context) {
+	var book models.Book
+	if err := c.ShouldBindJSON(&book); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+	}
+
+	doc, err := models.ElasticClient.Index().Index(INDEX_NAME).BodyJson(book).Do(c)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"status": "created",
+		"id":     doc.Id,
+	})
+}
+
+func UpdateBookTitleById(c *gin.Context) {
+	id := c.Param("id")
+
+	var book *models.Book
+	if err := c.ShouldBindJSON(&book); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+		return
+	}
+
+	_, err := models.ElasticClient.
```

Comment:
What if we will suddenly want to implement the book store search logic not using ElasticSearch?
We will have to rewrite this whole file.
Let's make it more generic - we can use an interface with all these methods and have handlers.go file not know the underlying implementation.
We can do this for the Redis too but in a separate interface, like we said it should be a cache that is autonomous to the logic of handling the request.